src/main/java/org/apache/sysds/runtime/matrix/data/FrameBlock.java

```diff
@@ -1865,6 +1865,39 @@ else if (!dataType.toString().equals(schemaString[i].trim()))
 		return this;
 	}
 
+	/*
+		This method validate the frame data against a column length constrain
+		if data value in any column is greater than specific threshold
+		output vector will store a 1 for that column position.
+		@param input MatrixBlock of valid lengths
+		@param output a boolean FrameBlock for valid/invalid features
+	 */
+	public FrameBlock invalidByLength(MatrixBlock feaLen) {
+		//sanity checks
+		if(this.getNumColumns() != feaLen.getNumColumns())
+			throw new DMLException("mismatch in number of columns in frame and corresponding feature-length vector");
+
+		ValueType[] outSchema = UtilFunctions.nCopies(this.getNumColumns(), ValueType.BOOLEAN);
+		String[][] outData = new String[this.getNumRows()][this.getNumColumns()];
+		FrameBlock outBlock = new FrameBlock(outSchema, outData);
+		outer: for (int i = 0; i < this.getNumColumns(); i++) {
```

Comment:
the explicit labels are unnecessary here because the referenced continue is the default behavior.